runelite-client/src/main/java/net/runelite/client/config/Keybind.java

```diff
@@ -125,9 +125,16 @@ protected boolean matches(KeyEvent e, boolean ignoreModifiers)
 			keyCode = KeyEvent.VK_UNDEFINED;
 		}
 
-		if (e.getID() == KeyEvent.KEY_RELEASED && keyCode != KeyEvent.VK_UNDEFINED)
+		if (e.getID() == KeyEvent.KEY_RELEASED)
 		{
-			return this.keyCode == keyCode;
+			if (keyCode != KeyEvent.VK_UNDEFINED)
+			{
+				return this.keyCode == keyCode;
+			}
+			else
+			{
+				return this.keyCode == keyCode && (this.modifiers & modifiers) == this.modifiers && ((mf & this.modifiers) == mf);
```

Comment:
how sure are you about the implicit mf unbox here? IJ warns for it, but I think for it to happen you have to get an VK_UNDEFINED key event which I am not sure happens?